apps/wallet-mobile/src/features/Swap/common/helpers.ts

```diff
@@ -32,3 +34,24 @@ export const getSellQuantityForLimitOrder = (
     sellTokenDecimals,
   ).toString() as BalanceQuantity
 }
+
+export const createYoroiEntry = (createOrder: Swap.CreateOrderData, address: string): YoroiEntry => {
+  const amountEntry = {}
+  const tokenId = createOrder?.amounts?.sell.tokenId
+  if (tokenId != null && createOrder.amounts.sell.quantity !== undefined) {
+    if (createOrder?.amounts?.sell.tokenId === '') {
```

Comment:
@SorinC6 I'd suggest split the code, to branch it as early as possible, `createEntryWithPrimaryToken`, `createEntryOnlySecondaryTokens`, since the math is different.